CSS/Indentation Rules.tmPreferences

```diff
@@ -36,7 +36,13 @@
 		<key>preserveIndent</key>
 		<false/>
 		<key>unIndentedLinePattern</key>
-		<string>^\s*(?:(?:/\*|\*+/).*)?$</string>
+		<string><![CDATA[(?x)
+			# standalone begin or end of block comments
+			#   ST doesn't apply `Indentation Rules - Comments`, because
+			#   comment scope doesn't cover the whole line. So they need
+			#   to be excluded from indentation engine here.
+			^\s* (?: (?: /\* | \*+/ ) .* )?$
```

Comment:
perhaps a nitpick, so feel free to ignore, but I believe for an unclosed `/*`, the `\n` would have the comment scope. And for `*/`, it seems a little odd (and redundant, because the regex pattern doesn't need to match the whole line) to check for `.*$` after it because it might be valid CSS code following a close comment marker which should still affect indentation, no?

Suggesting to use `^\s*$|\*/` instead, so that a comment opened on a prior line and which is closed on the current line would match this rule, with the caveat that it is untested, and perhaps the `^` is implied, and thus `^\s*$|.*\*/` would be necessary.

Reply:
It took that rule from...

https://github.com/sublimehq/Packages/blob/a423ed195aa8e39566ab84fca16b953b9ae78307/C%23/Indentation%20Rules.tmPreferences#L39-L40

The only thing I found not working was `**/`.

C++ seems to use something like that, too, even though it differs a little bit

https://github.com/sublimehq/Packages/blob/a423ed195aa8e39566ab84fca16b953b9ae78307/C%2B%2B/Indentation%20Rules.tmPreferences#L27-L28

I read about discussions how to solve indentation issues after block comments. Those pointed to this pattern, too.

The main issue with opening comment punctuation seems to be the non-comment scope of the whitespace in front of it, which causes indentation engine to apply `Indentation Rules` rather than `Indentation Rules - Comments` to that line. The newline and maybe the whole comment content probably doen't matter in that case, but due to existing implementations I took it as is.

---

What would you expect the last comment line to be indented like?

```css
/**
 * Test: Trailing comments spanning to next line
 * and containing closing brackets
 **/ .class { color: red; }

.class { color: red; }          /* }
*/
```

I think it needs to be ignored to keep comment formatation intact. As the whole block comment is not indented how should the indentation engine know know what the correct indentation level of the last line is, otherwise?

>  CSS code following a close comment marker which should still affect indentation, no?

Both Java and CSS contain several tests with block comments in between expressions. All of them seem to work.

> Suggesting to use `^\s*$|\*/` instead,

This breaks indentation tests.

```css
/**
 * Test: Complete trailing comment with braces
 **/

.class { color: red; }          /* ; } */
```

The content after block comments would be aligned with the closing `**/` and thus be indented one char too much.

---

I found several indentation rules which all treat the closing `*/` at the beginning of a line, but I am very suspicious about its meaning, function and usefullness.

https://github.com/sublimehq/Packages/blob/a423ed195aa8e39566ab84fca16b953b9ae78307/C%2B%2B/Indentation%20Rules.tmPreferences#L10

Neither indentation tests of Java nor CSS revealed any situation the leading `^ (.*\*/)?` has any useful effect. Maybe I miss some special situations, so we should add them to the indentation tests.

Reply:
A4 paper filled.

Reply:
Thanks for looking into it and the explanation :)

Reply:
It appears we can simplify the rule to `^\s* (?: /\* | \*+/ )`. Removing the trailing `.*$` doesn't break tests.